### PR TITLE
Random fixes

### PR DESCRIPTION
**Pull request by semoro**
DefaultArgumentsConversion
Temporary conversion for println

Support for new converter dedicated testData
Fix variables mutability processing
Expanded post-processings

### CODE REVIEW

File: j2k/newSrc/org/jetbrains/kotlin/j2k/conversions/DefaultArgumentsConversion.kt [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+        if (element !is JKClass) return recurse(element)
+
+        val methods = element.declarationList.filterIsInstance<JKMethod>().sortedBy { it.parameters.size }
+
+        checkMethod@ for (method in methods) {
+            val block = method.block as? JKBlock ?: continue
+            val singleStatement = block.statements.singleOrNull()
+
+            val call =
+                singleStatement.safeAs<JKExpressionStatement>()?.expression?.safeAs<JKMethodCallExpression>()
```
Comment (Dimach) [inline j2k/newSrc/org/jetbrains/kotlin/j2k/conversions/DefaultArgumentsConversion.kt:24]: Support this.someCall() case.

File: j2k/newSrc/org/jetbrains/kotlin/j2k/conversions/DefaultArgumentsConversion.kt [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+
+
+            // TODO: Filter by annotations, visibility, modality, modifiers like synchronized
+
+            for (i in method.parameters.indices) {
+                val parameter = method.parameters[i]
+                val targetParameter = calledMethod.parameters[i]
+                val argument = call.arguments.expressions[i]
+                if (parameter.name.value != targetParameter.name.value) continue@checkMethod
+                if (parameter.type.type != targetParameter.type.type) continue@checkMethod
```
Comment (Dimach) [inline j2k/newSrc/org/jetbrains/kotlin/j2k/conversions/DefaultArgumentsConversion.kt:41]: Same as above.

File: j2k/newSrc/org/jetbrains/kotlin/j2k/conversions/DefaultArgumentsConversion.kt [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+            val singleStatement = block.statements.singleOrNull()
+
+            val call =
+                singleStatement.safeAs<JKExpressionStatement>()?.expression?.safeAs<JKMethodCallExpression>()
+                        ?: singleStatement.safeAs<JKReturnStatement>()?.expression?.safeAs()
+                        ?: continue
+            val callee = call.identifier as? JKUniverseMethodSymbol ?: continue
+            val calledMethod = callee.target
+            if (calledMethod.parent != method.parent || callee.name != method.name.value) continue
+            if (calledMethod.returnType.type != method.returnType.type) continue
```
Comment (Dimach) [inline j2k/newSrc/org/jetbrains/kotlin/j2k/conversions/DefaultArgumentsConversion.kt:30]: Collapse thats if's to one.

File: j2k/newSrc/org/jetbrains/kotlin/j2k/conversions/DefaultArgumentsConversion.kt [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+            }
+
+
+            call.arguments.invalidate()
+            val defaults = call.arguments.expressions
+                .zip(calledMethod.parameters)
+                .drop(method.parameters.size)
+
+            for ((defaultValue, parameter) in defaults) {
+                parameter.initializer = defaultValue
```
Comment (Dimach) [inline j2k/newSrc/org/jetbrains/kotlin/j2k/conversions/DefaultArgumentsConversion.kt:52]: In defaultValue expression may be used other call argument. References must be updated.

**Review by Dimach — CHANGES_REQUESTED** [review]

**Comment by semoro** [comment]
Merged manually

**Pull request closed without merging**